### PR TITLE
Allowing fewer sensors

### PR DESCRIPTION
**Pull request by jeferal**
## Proposed changes

Making it possible for polhemus_ros_driver to have fewer sensors that the one a shadow glove has (6 per glove).
Story:
https://shadowrobot.atlassian.net/browse/SP-589

Confluence document on how to do this:
https://shadowrobot.atlassian.net/wiki/spaces/SDSR/pages/3433070607/Shadow+glove+with+fewer+sensors

Tested with unimanual left, right and bimanual.

## Checklist

Before posting a PR ensure that from each of the below categories **AT LEAST ONE BOX HAS BEEN CHECKED**. If more than one category is applicable then more can be checked. Also ensure that the proposed changes have been filled out with relevant information for reviewers.

## Tests

- [x] No tests required to be added. (For small changes that will be tested by CI/CD infrastructure).
- [ ] Added/Modified automated and PhantomHand CI tests (if a new class is added (Python or C++), the interface of that class must be unit tested).
- [ ] Manually tested in simulation (if simulation specific or no hardware required to test the functionality). 
- [x] Manually tested on hardware (if hardware specific or related).

## Documentation

- [ ] No documentation required to be added.
- [x] Added documentation (For any new feature, explain what it does and how to use it. Write the documentation in a relevant space, e.g. Github, Confluence, etc).
- [ ] Updated documentation (For changes to pre-existing features mentioned in the documentation).

### CODE REVIEW

File: include/polhemus_ros_driver/liberty.hpp [hunk 1]
```diff
@@ -39,7 +39,8 @@
 class Liberty : public Polhemus
 {
 public:
-    Liberty(std::string name, uint16_t rx_buffer_size, uint16_t tx_buffer_size);
+    Liberty(std::string name, uint16_t rx_buffer_size, uint16_t tx_buffer_size,
+            int sensors_right_glove, int sensors_left_glove);
```
Comment (bdenoun) [inline include/polhemus_ros_driver/liberty.hpp]: Do you really need an int? Can't this be an uint8?

File: include/polhemus_ros_driver/polhemus.hpp [hunk 2]
```diff
@@ -70,7 +69,8 @@ data_type_e;
 class Polhemus
 {
 public:
-  Polhemus(std::string name, uint16_t rx_buffer_size, uint16_t tx_buffer_size);
+  Polhemus(std::string name, uint16_t rx_buffer_size, uint16_t tx_buffer_size,
+           int sensors_right_glove, int sensors_left_glove);
```
Comment (bdenoun) [inline include/polhemus_ros_driver/polhemus.hpp]: Same question here

File: src/polhemus_ros_driver/polhemus.cpp [hunk 3]
```diff
@@ -37,9 +37,10 @@
 #endif
 
 
-Polhemus::Polhemus(std::string name, uint16_t rx_buffer_size, uint16_t tx_buffer_size):
+Polhemus::Polhemus(std::string name, uint16_t rx_buffer_size, uint16_t tx_buffer_size,
+                   int sensors_right_glove, int sensors_left_glove):
```
Comment (bdenoun) [inline src/polhemus_ros_driver/polhemus.cpp]: same here?

File: src/polhemus_ros_driver/polhemus_tf_broadcaster.cpp [hunk 4]
```diff
@@ -276,13 +276,12 @@ int main(int argc, char** argv)
 {
   libusb_device_handle *g_usbhnd = 0;
   vp_usbdevinfo g_usbinfo;
-  int i, nstations;
+  int i, nstations, sensors_right_glove, sensors_left_glove;
```
Comment (bdenoun) [inline src/polhemus_ros_driver/polhemus_tf_broadcaster.cpp]: could we have `i` and `nstations` more meaningful?

File: include/polhemus_ros_driver/liberty.hpp (hunk 1 above)
Comment (jeferal) [reply include/polhemus_ros_driver/liberty.hpp]: No, I don't really need it, thank you. I changed it in [ec70607](https://github.com/shadow-robot/polhemus_ros_driver/pull/55/commits/ec70607d039a8f69adcf42ab505f5fcef7b2ee86)

File: include/polhemus_ros_driver/polhemus.hpp (hunk 2 above)
Comment (jeferal) [reply include/polhemus_ros_driver/polhemus.hpp]: [ec70607](https://github.com/shadow-robot/polhemus_ros_driver/pull/55/commits/ec70607d039a8f69adcf42ab505f5fcef7b2ee86)

File: src/polhemus_ros_driver/polhemus.cpp (hunk 3 above)
Comment (jeferal) [reply src/polhemus_ros_driver/polhemus.cpp]: [ec70607](https://github.com/shadow-robot/polhemus_ros_driver/pull/55/commits/ec70607d039a8f69adcf42ab505f5fcef7b2ee86)

**Review by georgiablanco — APPROVED** [review]

File: src/polhemus_ros_driver/polhemus_tf_broadcaster.cpp (hunk 4 above)
Comment (jeferal) [reply src/polhemus_ros_driver/polhemus_tf_broadcaster.cpp]: Changed in [de26999](https://github.com/shadow-robot/polhemus_ros_driver/pull/55/commits/de269994fae3ac810d263dca8e5336b9cbfc67ea)

File: src/polhemus_ros_driver/viper.cpp [hunk 5]
```diff
@@ -415,10 +416,10 @@ bool Viper::persist_commands(void)
   }
 }
 
-int Viper::send_saved_calibration(int number_of_hands)
+int Viper::send_saved_calibration()
 {
   int retval = RETURN_ERROR;
```
Comment (bdenoun) [inline src/polhemus_ros_driver/viper.cpp]:
```suggestion
  int return_value = RETURN_ERROR;
```

File: src/polhemus_ros_driver/viper.cpp [hunk 6]
```diff
@@ -415,10 +416,10 @@ bool Viper::persist_commands(void)
...
-  retval = set_device_to_receive_saved_calibration(number_of_hands);
+  retval = set_device_to_receive_saved_calibration();
```
Comment (bdenoun) [inline src/polhemus_ros_driver/viper.cpp]:
```suggestion
  return_value = set_device_to_receive_saved_calibration();
```

File: src/polhemus_ros_driver/viper.cpp [hunk 7]
```diff
@@ -415,10 +416,10 @@ bool Viper::persist_commands(void)
...
   if (RETURN_ERROR == retval)
```
Comment (bdenoun) [inline src/polhemus_ros_driver/viper.cpp]:
```suggestion
  if (RETURN_ERROR == return_value)
```

Comment (bdenoun) [reply src/polhemus_ros_driver/viper.cpp]: Might have missed some

**Review by bdenoun — CHANGES_REQUESTED** [review]

**Comment by jeferal** [comment]
I changed the retval variable name for return_value in [d3662e4](https://github.com/shadow-robot/polhemus_ros_driver/pull/55/commits/d3662e401196eb1458bb59ea6c35af6e8a6d4610). Just in case, because of the last small changes, I tested the driver works by teleoperating unimanual and bimanual

**Review by bdenoun — APPROVED** [review]

**Pull request merged**